websocket_push_session.go

```diff
@@ -239,10 +290,39 @@ func (s *wsPushSession) listen() {
 					zap.String("sessionID", s.id),
 					zap.String("pushConfig", pushConfig.String()),
 				)
-				s.close(websocket.CloseUnsupportedData)
-				return
+
+				if s.handlesErrorEvents() {
+					s.setErrorState(true)
+					msgpack, json, err := prepareEventData(elemental.NewErrorEvent(elemental.Error{
+						Title:       "Bad request",
+						Subject:     "bahamut",
+						Description: fmt.Sprintf("unable to parse identity filters: %s", err),
+						Data: map[string]interface{}{
+							"attribute": "filters",
+						},
+					}, s.encodingWrite))
+
+					if err != nil {
+						zap.L().Error("unable to prepare error event encoding, closing socket", zap.Error(err))
```

Comment:
can you refactor this code in a function? it looks a bit redundant from the code in currentPushConfig

Reply:
Donezo ✅